### ISSUE TITLE
NVDA behaviour facing DOM update

### ISSUE THREAD

**Issue by hvinh** [nvaccess/nvda#7198]
### Steps to reproduce:
I was wondering if NVDA's behaviour in this case is working as designed.
I have a list of 2 buttons. When the first button is pressed, the list is refreshed:
- the list is recreated (new dom element)
- the first button is removed and only the second one remains

### Expected behavior:
After pressing tab, the focus goes to the second button and the button is announced

### Actual behavior:
After pressing tab, the focus goes to the second button but nothing is read

### System configuration:
NVDA version: 2017.1
Firefox version: 53.0.3

**Comment by jcsteh** [comment]
Please provide a simple test case demonstrating the problem. Thanks.

**Comment by hvinh** [comment]
Hello,
please follow the steps to reproduce: https://tinyurl.com/yd9cjvrg
1. select flights
2. fill travellers details
3. select seats for each passenger: one chargeable, one not chargeable
On the right side of the dialog, if we triggered the first trash icon to delete the chargeable seat, then the second trash icon will not be read on tab
![image](https://cloud.githubusercontent.com/assets/5523926/26642922/d9a91306-462f-11e7-88af-261329e81b0a.png)

Thanks & regards

**Comment by Adriani90** [comment]
@hvinh the link is not working on my side unfortunately. So I cannot test if this issue is still reproducible. are you still having it in NVDA 2019.1.1 and Firefox 66.0.3?

**Comment by hvinh** [comment]
Hello @Adriani90.

I do have the latest versions installed and i just confirmed that the issue is still happening.
Please use this link: https://tinyurl.com/y6spr8pq

You can follow the steps described previously, but select special assistance instead of seats (escort for the blind, wheelchair)

Then tab on the trash icon button for escort for the blind and delete it.
Tab again: nothing is read on the remaining trash icon button.

![image](https://user-images.githubusercontent.com/5523926/56274162-8ae90f00-60fe-11e9-99bf-4ac3f743aa1a.png)

Thanks & regards

**Comment by Adriani90** [comment]
Thanks for providing this link. I can reproduce this issue. in NVDA 2019.1.1 and Firefox 66.0.3. After checking two checkboxes for assistance (blind escort and wheel chair), I pressed tab until I heared "remove blind escort". I pressed enter. Then, I pressed tab to move to the second remove button for wheel chair. But NVDA didn't report anything. Following error is shown on the log file:

```
IO - inputCore.InputManager.executeGesture (23:34:11.328):
Input: kb(laptop):tab
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IA2Attributes (23:34:11.372):
IAccessibleObject.attributes COMError (-2147417848, 'Das aufgerufene Objekt wurde von den Clients getrennt.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (23:34:11.372):
accRole failed: (-2147417848, 'Das aufgerufene Objekt wurde von den Clients getrennt.', (None, None, None, 0, None))
```

In Chrome 73 there is the same problem.

cc: @jcsteh, @leonardder

**Comment by Adriani90** [comment]
I wonder if the trash buttons in the dialog are maybe inside of a div? This might cause NVDA to remain silent when tabbing.

**Comment by Adriani90** [comment]
cc: @feerrenrut

**Comment by Adriani90** [comment]
This might be closely related to #6606.

**Comment by Adriani90** [comment]
@hvinh again the last link is not working anymore. Can you still reproduce the issue in NVDA las alpha version? Please provide a test case.

You can download the last Alpha version of NVDA at this link:
https://www.nvaccess.org/files/nvda/snapshots/

**Comment by Adriani90** [comment]
We don't have an updated minimal test case here so it is impossible to investigate further. This might have been fixed by #14611.

If you are still having this issue, please comment with a minimal test case and we can reopen.
I am closing this issue as abandoned since we didn't get any updates from the initial author after request since more than 5 years.
